samples/CommunityToolkit.Maui.Sample/Pages/Views/Maps/MapsPinsPage.xaml.cs

```diff
@@ -0,0 +1,115 @@
+using Microsoft.Maui.Controls.Maps;
+using CommunityToolkit.Maui.Sample.ViewModels.Views;
+using Microsoft.Maui.Maps;
+
+namespace CommunityToolkit.Maui.Sample.Pages.Views;
+
+public partial class MapsPinsPage : BasePage<MapsPinsViewModel>
+{
+	private readonly Random locationRandomSeed = new();
+	private int locationIncrement = 0;
+
+	// TODO generate actual random pins
+	private readonly Location[] randomLocations =
+	{
+			new Location(51.8833333333333, 176.65),
+			new Location(21.3166666666667, 157.833333333333),
+			new Location(71.3, 156.766666666667),
+			new Location(19.7, 155.083333333333),
+			new Location(61.2166666666667, 149.9),
+			new Location(70.2, 148.516666666667),
+			new Location(64.85, 147.716666666667),
+			new Location(57.05, 135.333333333333),
+			new Location(60.7166666666667, 135.05),
+			new Location(58.3, 134.416666666667),
+			new Location(69.45, 133.033333333333),
+			new Location(48.4333333333333, 123.366666666667),
+			new Location(49.25, 123.1),
+			new Location(45.5166666666667, 122.683333333333),
+			new Location(37.7833333333333, 122.416666666667),
+			new Location(47.6166666666667, 122.333333333333),
+			new Location(38.55, 121.466666666667),
+			new Location(50.6833333333333, 120.333333333333),
+			new Location(39.5333333333333, 119.816666666667),
+			new Location(34.4333333333333, 119.716666666667),
+			new Location(49.8833333333333, 119.5),
+			new Location(55.1666666666667, 118.8),
+			new Location(34.05, 118.25),
+			new Location(33.95, 117.4),
+			new Location(32.7166666666667, 117.166666666667),
+			new Location(32.5333333333333, 117.033333333333),
+			new Location(31.85, 116.6),
+			new Location(43.6166666666667, 116.2),
+			new Location(32.6666666666667, 115.466666666667),
+			new Location(36.1833333333333, 115.133333333333),
+			new Location(62.45, 114.4),
+			new Location(51.05, 114.066666666667),
+			new Location(53.5333333333333, 113.5),
+			new Location(33.45, 112.066666666667),
+			new Location(46.6, 112.033333333333),
+		};
+
+	public MapsPinsPage(MapsPinsViewModel mapsPinsViewModel) : base(mapsPinsViewModel)
+	{
+		InitializeComponent();
+	}
+
+	protected override void OnNavigatedTo(NavigatedToEventArgs args)
+	{
+		base.OnNavigatedTo(args);
+
+		var microsoftLocation = new Location(47.64232, -122.13684);
+
+		var microsoftPin = new Pin()
+		{
+			Address = "One Microsoft Way, Redmond, USA",
+			Label = "Microsoft Visitors Center",
+			Location = microsoftLocation,
+		};
+
+		microsoftPin.MarkerClicked += (s, a) =>
+		{
+			DisplayAlert("Marker", "OK", "OK");
+		};
+
+		microsoftPin.InfoWindowClicked += (s, a) =>
+		{
+			DisplayAlert("Info", "OK", "OK");
+		};
+
+		pinsMap.Pins.Add(microsoftPin);
+
+		pinsMap.MoveToRegion(MapSpan.FromCenterAndRadius(microsoftLocation, Distance.FromKilometers(1)));
+	}
```

Comment:
I'm not sure if this works for me. I end up looking at Japan or something and don't see a pin at the Microsoft location. Probably because the map isn't loaded at that time? 

So let's either remove this, or implement it in a way that works so people know how to do this.